lib/Twig/Extensions/TokenParser/Trans.php

```diff
@@ -69,6 +73,27 @@ public function getTag()
         return 'trans';
     }
 
+    /**
+     * Overrides Twig_TokenParser::setParser.
+     * Sets the parser associated with this token parser and extract
+     * configuration from the parser environment.
+     *
+     * @param Twig_Parser $parser A Twig_Parser instance
+     */
+    public function setParser(Twig_Parser $parser)
+    {
+        parent::setParser($parser);
+
+        // Hack to allow to read the private environment from the $parser.
+        // I don't get why it was decided to deprecate the "getEnvironment"
+        // from the parser, but... whatever.
```

Comment:
Please don't do that. There is 0 backward compatibility promise for that, so it may break in any release (even in patch releases)

Reply:
Absolutely, it was kind of a hack. I've transformed this pattern into a `Twig\Extensions\Node\Trans\Options` object that is fully passed down to the `TokenParser` and the `Node\Trans` itself as you suggested. Thank you!